client/signup/steps/jpo-summary/index.jsx

```diff
@@ -169,19 +209,57 @@ class JPOSummaryStep extends React.Component {
 					</tbody>
 				</table>
 				<div>
-					<Button primary onClick={ this.completeOnboarding }>
+					<a
+						className="jpo-summary__visit-site button is-primary"
+						href={ get( this.props.signupDependencies, [ 'jpoConnect', 'queryObject', 'home_url' ] ) }>
 						{
-							translate( 'Visit your site' )
+							translate( 'Visit site' )
 						}
-					</Button>
+					</a>
 				</div>
 			</div>
 		);
 	}
 
+	componentWillMount() {
+		let choices = this.getOnboardingChoices();
+		const siteId = get( this.props.signupDependencies, [ 'jpoConnect', 'queryObject', 'client_id' ], -1 ),
+			hasBusinessInfo = some( choices.onboarding.businessInfo, i => '' !== i );
+		if ( 'store' === choices.onboarding.genre ) {
+			this.props.installPlugin( siteId, { slug: 'woocommerce', id: 'woocommerce/woocommerce' } );
+		}
+		if ( hasBusinessInfo ) {
+			choices = extend( {}, choices, {
+				widgets: true,
+			} );
+		}
+		this.props.updateSettings(
+			siteId,
+			choices,
+			false // don't sanitize settings
+		).then( () => {
+			if ( hasBusinessInfo ) {
+				const {
+					businessName,
+					businessAddress,
+					businessCity,
+					businessState,
+					businessZipCode
+				} = choices.onboarding.businessInfo;
+				this.props.addWidget( siteId, {
+					id_base: 'widget_contact_info',
+					settings: {
+						title: businessName,
+						address: businessAddress + '\n' + businessCity + '\n' + businessState + '\n' + businessZipCode,
```

Comment:
This could be `.join`ed. `[ businessAddress, businessCity, businessState, businessZipCode].join( '\n' ). Not that it's bad as it is right now